x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -69,35 +68,45 @@ public Realms(Settings settings, Environment env, Map<String, Realm.Factory> fac
         this.threadContext = threadContext;
         this.reservedRealm = reservedRealm;
         assert factories.get(ReservedRealm.TYPE) == null;
-        this.realms = initRealms();
-        // pre-computing a list of internal only realms allows us to have much cheaper iteration than a custom iterator
-        // and is also simpler in terms of logic. These lists are small, so the duplication should not be a real issue here
-        List<Realm> standardRealms = new ArrayList<>();
-        List<Realm> nativeRealms = new ArrayList<>();
-        for (Realm realm : realms) {
-            // don't add the reserved realm here otherwise we end up with only this realm...
-            if (InternalRealms.isStandardRealm(realm.type())) {
-                standardRealms.add(realm);
-            }
 
-            if (FileRealmSettings.TYPE.equals(realm.type()) || NativeRealmSettings.TYPE.equals(realm.type())) {
-                nativeRealms.add(realm);
-            }
-        }
+        this.allConfiguredRealms = initRealms();
+        this.allConfiguredRealms.forEach(r -> r.initialize(this, licenseState));
+        this.activeRealms = allConfiguredRealms;
```

Comment:
Why not call `recomputeActiveRealms` here? Is it because licenseState may not be ready at this point? It also assumes that the first usage of `activeRealms` will be after the licenseState is ready (and listener invoked). Will it be possible that somehow an unlicensed realm is enabled because of racing condition?